src/NuGet.Core/NuGet.Common/SystemEnvironmentVariableProvider.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.Common
+{
+    /// <summary>
+    /// Represents an implementation of <see cref="IEnvironmentVariableProvider" /> that manipulates environment variables on the current system.
+    /// </summary>
+    public sealed class SystemEnvironmentVariableProvider : IEnvironmentVariableProvider
+    {
+        private SystemEnvironmentVariableProvider()
+        {
+        }
+
+        /// <summary>
+        /// Gets an <see cref="IEnvironmentVariableProvider" /> that manipulates environment variables on the current system.
+        /// </summary>
+        public static IEnvironmentVariableProvider Instance { get; } = new SystemEnvironmentVariableProvider();
+
+#pragma warning disable RS0030 // Do not use banned APIs
+        public string? GetEnvironmentVariable(string variable) => Environment.GetEnvironmentVariable(variable);
+#pragma warning restore RS0030 // Do not use banned APIs
+
+#pragma warning disable RS0030 // Do not use banned APIs
+        public string? GetEnvironmentVariable(string variable, EnvironmentVariableTarget target) => Environment.GetEnvironmentVariable(variable, target);
+#pragma warning restore RS0030 // Do not use banned APIs
+
+#pragma warning disable RS0030 // Do not use banned APIs
+        public void SetEnvironmentVariable(string name, string? value) => Environment.SetEnvironmentVariable(name, value);
+#pragma warning restore RS0030 // Do not use banned APIs
+
+#pragma warning disable RS0030 // Do not use banned APIs
+        public void SetEnvironmentVariable(string name, string? value, EnvironmentVariableTarget target) => Environment.SetEnvironmentVariable(name, value, target);
+#pragma warning restore RS0030 // Do not use banned APIs
```

Comment:
```suggestion
#pragma warning disable RS0030 // Do not use banned APIs
        public string? GetEnvironmentVariable(string variable) => Environment.GetEnvironmentVariable(variable);

        public string? GetEnvironmentVariable(string variable, EnvironmentVariableTarget target) => Environment.GetEnvironmentVariable(variable, target);

        public void SetEnvironmentVariable(string name, string? value) => Environment.SetEnvironmentVariable(name, value);
        
        public void SetEnvironmentVariable(string name, string? value, EnvironmentVariableTarget target) => Environment.SetEnvironmentVariable(name, value, target);
#pragma warning restore RS0030 // Do not use banned APIs
```

Reply:
I had it both ways when I was developing this.  The only problem with your suggestion is that it lets anyone add new code between the `#prama` that gets to use the suppression.  I know all of these suppressions are ugly in code but it makes sure that anyone adding to this class has to think about any code analysis warnings they might have introduced.  That said, I'm fine with either one, this is just what I personally settled on.  What does everyone else think?